From 993cd377ba7f9c2e82c5e7dd4ac1a9fcca689c7c Mon Sep 17 00:00:00 2001
From: Philipp von Weitershausen <philikon@silklabs.com>
Date: Thu, 16 Jun 2016 17:53:52 -0700
Subject: [PATCH] Build fixes

- Only build armv7 and x86 against libsodium binaries already in tree
- Don't force-install build dependencies such as the Android SDK and NDK.
- Don't depend on guava, as it isn't necessary but bloats the build quite substantively.

With swig installed, ANDROID_SDK and ANDROID_NDK env vars set, and libsodium binaries
prebuilt in the tree, all it takes to build the binary (JNI) parts of libsodium-jni
is

  ./build-libsodiumjni.sh
---
 build-libsodiumjni.sh |  7 ++++---
 build.gradle          |  3 +--
 jni/Android.mk        | 23 +++++++++++------------
 jni/Application.mk    |  2 +-
 pom.xml               |  5 -----
 5 files changed, 17 insertions(+), 23 deletions(-)

diff --git a/build-libsodiumjni.sh b/build-libsodiumjni.sh
index c6104f3..1a1c419 100755
--- a/build-libsodiumjni.sh
+++ b/build-libsodiumjni.sh
@@ -1,8 +1,9 @@
 #!/bin/bash -ev
 
-. ./setenv.sh
+pushd jni
+swig -java -package org.libsodium.jni -outdir ../src/main/java/org/libsodium/jni sodium.i
+popd
 
-ndk-build
+$ANDROID_NDK/ndk-build
 rm -rf src/main/jniLibs/
 cp -R libs src/main/jniLibs
-gradle build
diff --git a/build.gradle b/build.gradle
index 996522f..d38fd05 100644
--- a/build.gradle
+++ b/build.gradle
@@ -8,7 +8,7 @@ repositories {
 
 android {
     compileSdkVersion 23
-    buildToolsVersion "23.0.2"
+    buildToolsVersion "23.0.3"
 
     compileOptions {
         sourceCompatibility JavaVersion.VERSION_1_7
@@ -44,7 +44,6 @@ dependencies {
     // TODO are these required in this lib project?
     //compile fileTree(include: ['*.jar'], dir: 'libs')
    //compile 'com.android.support:appcompat-v7:23.1.1'
-   compile group: 'com.google.guava', name: 'guava', version: '19.0'
    testCompile 'junit:junit:4.12'
 }
 
diff --git a/jni/Android.mk b/jni/Android.mk
index 4ce28f5..8622a01 100644
--- a/jni/Android.mk
+++ b/jni/Android.mk
@@ -14,12 +14,9 @@
 #
 LOCAL_PATH := $(call my-dir)
 
-# Bugfix for arm, which should refer to the armv6 folder
-# Bugfix for x86, which should refer to the i686 folder
-# Bugfix for mips, which should refer to the mips32 folder
 MY_ARCH_FOLDER := $(TARGET_ARCH)
 ifeq ($(MY_ARCH_FOLDER),arm)
-    MY_ARCH_FOLDER = armv6
+    MY_ARCH_FOLDER = armv7-a
 endif
 ifeq ($(MY_ARCH_FOLDER),x86)
     MY_ARCH_FOLDER = i686
@@ -28,23 +25,25 @@ ifeq ($(MY_ARCH_FOLDER),mips)
     MY_ARCH_FOLDER = mips32
 endif
 
+external_dir := $(realpath $(LOCAL_PATH)/../..)
 
 include $(CLEAR_VARS)
 LOCAL_MODULE := sodium
-LOCAL_SRC_FILES := ../libsodium/libsodium-android-$(MY_ARCH_FOLDER)/lib/libsodium.a #/installs/libsodium/libsodium-android-(x86|arm|mips)/lib/libsodium.a
+LOCAL_SRC_FILES := \
+  $(external_dir)/libsodium-android-$(MY_ARCH_FOLDER)/lib/libsodium.a \
+
 include $(PREBUILT_STATIC_LIBRARY)
 
-include $(CLEAR_VARS)
 
+include $(CLEAR_VARS)
 LOCAL_MODULE    := sodiumjni
-LOCAL_SRC_FILES :=  \
-sodium_wrap.c
+LOCAL_SRC_FILES := \
+  sodium_wrap.c \
 
 LOCAL_CFLAGS   += -Wall -g -pedantic -std=c99
+LOCAL_C_INCLUDES += \
+  $(external_dir)/libsodium-android-$(MY_ARCH_FOLDER)/include \
+  $(external_dir)/libsodium-android-$(MY_ARCH_FOLDER)/include/sodium \
 
-LOCAL_C_INCLUDES += ../libsodium/libsodium-android-$(MY_ARCH_FOLDER)/include ../libsodium/libsodium-android-$(MY_ARCH_FOLDER)/include/sodium /usr/local/include
 LOCAL_STATIC_LIBRARIES += android_native_app_glue sodium
-#LOCAL_LDLIBS += -llog -lsodium
-
-
 include $(BUILD_SHARED_LIBRARY)
diff --git a/jni/Application.mk b/jni/Application.mk
index 03cb879..32e2601 100644
--- a/jni/Application.mk
+++ b/jni/Application.mk
@@ -1,3 +1,3 @@
 APP_STL := stlport_static
 
-APP_ABI := armeabi armeabi-v7a x86 mips
\ No newline at end of file
+APP_ABI := armeabi-v7a x86
\ No newline at end of file
diff --git a/pom.xml b/pom.xml
index 871064e..c09fe25 100644
--- a/pom.xml
+++ b/pom.xml
@@ -51,11 +51,6 @@
 
     <dependencies>
         <dependency>
-            <groupId>com.google.guava</groupId>
-            <artifactId>guava</artifactId>
-            <version>19.0</version>
-        </dependency>
-        <dependency>
             <groupId>junit</groupId>
             <artifactId>junit</artifactId>
             <version>4.11</version>
-- 
2.8.3

